### ISSUE TITLE
[Bug]: Run `timescaledb_post_restore()` twice to get jobs running

### ISSUE THREAD

**Issue by Harkishen-Singh** [timescale/timescaledb#5842]
### What type of bug is this?

Unexpected error

### What subsystems and features are affected?

Policy

### What happened?

I was performing a series of migration tests from MST to Timescale Cloud. The migration database contained 202 jobs. Before the migration, I executed `timescaledb_pre_restore()` and after the migration completed, I executed `timescaledb_post_restore()`. 

I think I received the following error logs in Timescale Cloud (I might be wrong as I went back in time to find these):
```shell
2023-06-22 14:35:07 UTC [12329]: ERROR: tried calling catalog_get when extension isn't loaded
2023-06-22 14:35:07 UTC [115]: LOG: background worker "TimescaleDB Background Worker Scheduler" (PID 12329) exited with exit code 1
```
The entries in `timescaledb_information.job_stats` had 2 entries only, `Telemetry Reporter` & `Error ..`.
After 20 mins, the `job_stats` had the same entries.

So, I decided to execute `timescaledb_post_restore()` again. As soon as I completed, I saw 40 jobs under `job_stats` view, and after waiting for 15 mins, I can see all the 202 jobs running on Timescale Cloud. In the logs, I can see TimescaleDB background worker running. 

I have confirmed the above behaviour in several migration tests.

### TimescaleDB version affected

2.11.0

### PostgreSQL version used

15

### What operating system did you use?

Timescale Cloud

### What installation method did you use?

Not applicable

### What platform did you run on?

Timescale Cloud

### How can we reproduce the bug?

```bash
- Migrate any database that has some policies -- retention, reordering, caggs refresh policies; to Cloud, with `timescaledb_pre_restore()` executed before the process.

- After the migration complete, check the number of  entries in `timescaledb_information.job_stats` joined with `timescaledb_information.jobs`

- Run `timescaledb_post_restore()`

- Wait for sometime

- Check the entries in `timescaledb_information.job_stats` joined with `timescaledb_information.jobs`. You should have lesser rows than expected.

- Run `timescaledb_post_restore()` again

- Wait for sometime

- Check the entries in `timescaledb_information.job_stats` joined with `timescaledb_information.jobs`. You should see the expected number of rows.
```

**Comment by fabriziomello** [comment]
@Harkishen-Singh can u please provide a reproducible (step by step using SQL scripts) test case. Is hard to follow exactly what you're doing.

**Comment by Harkishen-Singh** [comment]
@fabriziomello sorry for the delay in response.

The steps are same as what we suggest in the [docs](https://docs.timescale.com/use-timescale/latest/migration/pg-dump-and-restore/#restore-your-entire-database-from-backup). The problem happens when we run `timescaledb_post_restore()`. I have to run the post-restore function twice to get the jobs running.

Basically, in Timescale (Cloud)
1. `select timescaledb_pre_restore()`
2. Do the migration of data using pg_restore
3. `select timescaledb_post_restore()`

Check the logs for any error. The `timescaledb_information.job_stats` will not have any new job stats.

Do `timescaledb_post_restore()` again. Now you should see entries in the new job stats.

**Comment by jnidzwetzki** [comment]
Hello @Harkishen-Singh,

Thanks for reporting this problem. I was able to reproduce the problem locally (using timescaledb 2.12.0-dev) with one hypertable and one compression policy.  After executing `SELECT timescaledb_pre_restore();` I also faced the error `tried calling catalog_get when extension isn't loaded` in the logfile.

```
# Create a test database
createdb testbackup
psql testbackup

CREATE EXTENSION timescaledb;

CREATE TABLE sensor_data(
time timestamp NOT NULL,
sensor_id integer NOT NULL,
cpu double precision,
temperature double precision);

SELECT FROM create_hypertable(relation=>'sensor_data', time_column_name=> 'time');

ALTER TABLE sensor_data SET (timescaledb.compress, timescaledb.compress_orderby = 'sensor_id, cpu, temperature');

INSERT INTO sensor_data
SELECT time + (INTERVAL '1 minute' * random()) AS time,
sensor_id, 
random() AS cpu, 
random()* 100 AS temperature
FROM
generate_series(now() - INTERVAL '1 months', now() - INTERVAL '1 week', INTERVAL '100 minute') AS g1(time),
generate_series(1, 2, 1 ) AS g2(sensor_id)
ORDER BY time;

SELECT add_compression_policy('sensor_data', INTERVAL '8 weeks', schedule_interval => '1m');

testbackup=# SELECT last_run_started_at,total_runs from timescaledb_information.job_stats where job_id = 1000;
     last_run_started_at      | total_runs 
------------------------------+------------
 2023-09-13 14:53:47.07208+02 |          1
(1 row)

# Backup
pg_dump -Fc -f testbackup.bak testbackup

# Restore
dropdb testbackup
createdb testbackup

psql testbackup
CREATE EXTENSION IF NOT EXISTS timescaledb;
SELECT timescaledb_pre_restore();

\! pg_restore -Fc -d testbackup testbackup.bak

SELECT timescaledb_post_restore();

# Logfile
2023-09-13 15:08:58.210 CEST [2521755] LOG:  background worker "TimescaleDB Background Worker Scheduler" (PID 2523019) exited with exit code 1
2023-09-13 15:09:05.682 CEST [2523030] ERROR:  tried calling catalog_get when extension isn't loaded
2023-09-13 15:09:05.686 CEST [2521755] LOG:  background worker "TimescaleDB Background Worker Scheduler" (PID 2523030) exited with exit code 1
```
